source/dub/generators/generator.d

```diff
@@ -97,6 +97,7 @@ class ProjectGenerator
 			BuildSettings buildSettings;
 			auto config = configs[pack.name];
 			buildSettings.processVars(m_project, pack, pack.getBuildSettings(settings.platform, config), settings, true);
+			if (settings.buildSettings.options & BuildOption.lowmem) buildSettings.options |= BuildOption.lowmem;
```

Comment:
adding on what we previously discussed: the lowmem addition could be copied into processVars inside the `if (include_target_settings)` while still keeping the functionality of the function sane.

Below where it calls finalizeGeneration right now it doesn't include the lowmem flag for example. Even though it isn't used right now, it might be subject to change in the future (because it processes the environment variables in there)

What would you think about that approach?

Reply:
I rather put it in its current location, being careful that the lowmem only applies to the build and not to finalizeGeneration/performPostGenerateActions.
The reason for doing so is that I didn't see the vision of lowmem being used in finalizeGeneration/performPostGenerateActions.
I made it a priority to minimize the impact.

Also, there still doesn't seem to be a sensible reason to put this process in a function for variable expansion.
From this perspective, we thought it was reasonable to handle it outside of the function.